src/analysis/normed_space/star/multiplier.lean

```diff
@@ -0,0 +1,414 @@
+  and show it is complete
++ show that the image of `A` in `𝓜(𝕜, A)` is an essential ideal
++ prove the universal property of `𝓜(𝕜, A)`
+* Construct a double centralizer from a pair of maps `L : A → A`, `R : A → A` satisfying the
+  centrality condition `∀ x y, R x * y = x * L y`.
+-/
+
+noncomputable theory
+
+open_locale nnreal ennreal
+open nnreal continuous_linear_map
+
+universes u v
+
+/-- The type of *double centralizers*, also known as the *multiplier algebra* and denoted by
+`𝓜(𝕜, A)`, of a non-unital normed algebra. -/
+@[ext]
+structure double_centralizer (𝕜 : Type u) (A : Type v) [nontrivially_normed_field 𝕜]
+  [non_unital_normed_ring A] [normed_space 𝕜 A] [smul_comm_class 𝕜 A A] [is_scalar_tower 𝕜 A A] :=
+(left : A →L[𝕜] A)
+(right : A →L[𝕜] A)
+(central : ∀ x y : A, right x * y = x * left y)
+
+localized "notation `𝓜(` 𝕜 `, ` A `)` := double_centralizer 𝕜 A" in multiplier_algebra
+
+namespace double_centralizer
+
+section nontrivially_normed
+
+variables (𝕜 A : Type*) [nontrivially_normed_field 𝕜] [non_unital_normed_ring A]
+variables [normed_space 𝕜 A] [smul_comm_class 𝕜 A A] [is_scalar_tower 𝕜 A A]
+
+instance : inhabited 𝓜(𝕜, A) :=
+{ default := ⟨1, 1, by simp only [one_apply, eq_self_iff_true, forall_const]⟩ }
+
+/-!
+### Normed space structure
+
+Because the multiplier algebra is defined as the algebra of double centralizers, there is a natural
+map `double_centralizer.prod_mk := λ a, (a.left, a.right) : 𝓜(𝕜, A) → (A →L[𝕜] A) × (A →L[𝕜] A)`.
+We use this map to pull back the normed space structure from `(A →L[𝕜] A) × (A →L[𝕜] A)` to
+`𝓜(𝕜, A)`, which provides a definitional isometric embedding. Consequently, completeness of
+`𝓜(𝕜, A)` is obtained by proving that the range of this map is closed.
+-/
+
+/-- the canonical map of `𝓜(𝕜, A)` into `(A →L[𝕜] A) × (A →L[𝕜] A)`. -/
+@[simp] def prod_mk (a : 𝓜(𝕜, A)) : (A →L[𝕜] A) × (A →L[𝕜] A) := (a.left, a.right)
+
+variables {𝕜 A}
+
+lemma injective_prod_mk : function.injective (prod_mk 𝕜 A) :=
+λ a b h, ext a b (prod.ext_iff.mp h).1 (prod.ext_iff.mp h).2
+
+lemma range_prod_mk : set.range (prod_mk 𝕜 A) = {lr | ∀ x y, lr.2 x * y = x * lr.1 y} :=
+set.ext $ λ x, ⟨by {rintro ⟨a, rfl⟩, exact a.central}, λ hx, ⟨⟨x.1, x.2, hx⟩, by simp⟩⟩
+
+instance : has_add 𝓜(𝕜, A) :=
+{ add := λ a b,
+  { left := a.left + b.left,
+    right := a.right + b.right,
+    central := λ x y, by simp only [continuous_linear_map.add_apply, add_mul, mul_add, central] } }
+
+instance : has_zero 𝓜(𝕜, A) :=
+{ zero :=
+  { left := 0,
+    right := 0,
+    central := λ x y, by simp only [continuous_linear_map.zero_apply, zero_mul, mul_zero] } }
+
+instance : has_neg 𝓜(𝕜, A) :=
+{ neg := λ a,
+  { left := -(a.left),
+    right := -(a.right),
+    central := λ x y, by simp only [continuous_linear_map.neg_apply, neg_mul,
+                      mul_neg, neg_inj, central]}}
+
+instance : has_sub 𝓜(𝕜, A) :=
+{ sub := λ a b,
+  { left := a.left - b.left,
+    right := a.right - b.right,
```

Comment:
To illustrate my suggestion of extending `(A →L[𝕜] A) × (A →L[𝕜] A)`, these algebraic operations can be defined as follows:
```suggestion
  { to_prod := a.to_prod - b.to_prod,
```
which gives you a bunch of lemmas for free.